tests/elemental/installation.pm

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Test boot of Elemental OS
+# Maintainer: elemental@suse.de
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+
+use testapi;
+use power_action_utils qw(power_action);
+use serial_terminal qw(select_serial_terminal);
+
+sub run {
+    my ($self) = @_;
+    my $cloudconfig = "/tmp/cloud-config.yaml";
+    my %boot_entries = (
+        active => {
+            tag => 'elemental-bootmenu-default'
+        },
+        passive => {
+            tag => 'elemental-bootmenu-fallback'
+        },
+        recovery => {
+            tag => 'elemental-bootmenu-recovery'
+        }
+    );
+
+
+    # Wait for GRUB
+    $self->wait_grub(bootloader_time => 120);
+    send_key('ret', wait_screen_change => 1);
+    wait_still_screen(timeout => 90);
+    save_screenshot();
+
+    ## No GUI, easier and quicker to use the serial console
+    select_serial_terminal();
+
+    # Add a simple cloud-config
+    my $rootpwd = script_output('openssl passwd -6 ' . get_var('PASSWORD'));
```

Comment:
No idea exactly how Elemental is supposed to work, so this may be a non-issue, but if by any chance this can leave infrastructure in the cloud you want to be protected, it may be better to use something like:
```
get_var('_SECRET_PASSWORD');
```
As it will not be published as part of the test results.

Usually what I do is `get_var('_SECRET_PASSWORD', get_var('PASSWORD'))` so any of the two can work.

Reply:
> Couldn't see the verification run as it seems to be tied to your machine connected to the VPN perhaps?

Yes, I will anyway run a new VR before merging to validate again.

As for the password it's only the root password generated live for the deployed OS, nothing secret. But maybe I can use a different variable name to be less generic.

Reply:
I now remember why I used `PASSWORD` variable: it's because I have a default password set in the ISO for the first boot (set to `ros`) and to use this one for the `select_serial_console` function I need to set the value in `PASSWORD`.

It is defined in `set_defaults_for_username_and_password` function available in `lib/main_common.pm` to be able to use it.

So, as for the boot-after-installation test I need also to be able to connect I have to define a root password with `cloud-init` and so to simplify my test code I simply use the same `ros` value after installation. This is why `PASSWORD` is used.

Reply:
But as this is not so good to use the same password I will push a modification to use a new one after OS installation.

Reply:
Done!